### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Anna-Kuzmenko**
[DEMO LINK](https://Anna-Kuzmenko.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,121 @@
+import React, { useState } from 'react';
```
Comment (glots-a) [inline src/App.tsx:1]: Where is 404 problem with DEMO LINK

**Review by glots-a — APPROVED** [review]
very nice!

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,121 @@
...
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
```
Comment (SerhiiKirik) [inline src/App.tsx:36]: move this to the separate function

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,121 @@
...
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
```
Comment (SerhiiKirik) [inline src/App.tsx:11]: Don't forget about past checklists
```suggestion
function getUserById(id: number): User | null {
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,121 @@
...
+          if (!selectedUserName) {
+            setIsUserChose(false);
+          }
+
+          if (newTitle && selectedUserName) {
+            const newUser = usersFromServer
+              .find(user => user.name === selectedUserName) || null;
+
+            const todosId = visibleTodos.map(todo => todo.id);
+            const newTodoId = Math.max(...todosId) + 1;
```
Comment (SerhiiKirik) [inline src/App.tsx:52]: create a separate function for getNewId

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,121 @@
...
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
+          <input
+            id="title"
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            name="title"
+            value={newTitle}
+            onChange={(event) => {
```
Comment (SerhiiKirik) [inline src/App.tsx:78]: make a separate function for this

File: src/App.tsx [hunk 6]
```diff
@@ -1,61 +1,121 @@
...
+              setNewTitle(event.target.value);
+              setIsTitle(true);
+            }}
+          />
+          {!isTitle && (
+            <span className="error">
+              Please enter a title
+            </span>
+          )}
```
Comment (SerhiiKirik) [inline src/App.tsx:87]: in all cases
```suggestion
          <label htmlFor="title">Title: </label>
          
          <input
            id="title"
            type="text"
            data-cy="titleInput"
            placeholder="Enter a title"
            name="title"
            value={newTitle}
            onChange={(event) => {
              setNewTitle(event.target.value);
              setIsTitle(true);
            }}
          />
          
          {!isTitle && (
            <span className="error">
              Please enter a title
            </span>
          )}
```

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Well done 🔥 
Let's deploy your solution & fix some problem 🧑‍💻

**Pull request closed without merging**